config/manifests/preview/secrets-store-csi-driver-operator.clusterserviceversion.yaml

```diff
@@ -0,0 +1,366 @@
+    createdAt: "2023-06-12T00:00:00Z"
+    description: Install and configure Secrets Store CSI driver.
+    olm.skipRange: ">=4.13.0-0 <4.14.0"
+  labels:
+    operator-metering: "true"
+    "operatorframework.io/arch.amd64": supported
+    "operatorframework.io/arch.arm64": supported
+spec:
+  displayName: Secrets Store CSI Driver Operator
+  description: >
+    Operator that installs and configures the CSI driver for Secrets Store.
+    The Secrets Store CSI Driver Operator is a Technology Preview feature.
+    For more information on Red Hat Technology Preview features support scope, see https://access.redhat.com/support/offerings/techpreview/.
+  icon:
+    - base64data: PHN2ZyBpZD0iTGF5ZXJfMSIgZGF0YS1uYW1lPSJMYXllciAxIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAxOTIgMTQ1Ij48ZGVmcz48c3R5bGU+LmNscy0xe2ZpbGw6I2UwMDt9PC9zdHlsZT48L2RlZnM+PHRpdGxlPlJlZEhhdC1Mb2dvLUhhdC1Db2xvcjwvdGl0bGU+PHBhdGggZD0iTTE1Ny43Nyw2Mi42MWExNCwxNCwwLDAsMSwuMzEsMy40MmMwLDE0Ljg4LTE4LjEsMTcuNDYtMzAuNjEsMTcuNDZDNzguODMsODMuNDksNDIuNTMsNTMuMjYsNDIuNTMsNDRhNi40Myw2LjQzLDAsMCwxLC4yMi0xLjk0bC0zLjY2LDkuMDZhMTguNDUsMTguNDUsMCwwLDAtMS41MSw3LjMzYzAsMTguMTEsNDEsNDUuNDgsODcuNzQsNDUuNDgsMjAuNjksMCwzNi40My03Ljc2LDM2LjQzLTIxLjc3LDAtMS4wOCwwLTEuOTQtMS43My0xMC4xM1oiLz48cGF0aCBjbGFzcz0iY2xzLTEiIGQ9Ik0xMjcuNDcsODMuNDljMTIuNTEsMCwzMC42MS0yLjU4LDMwLjYxLTE3LjQ2YTE0LDE0LDAsMCwwLS4zMS0zLjQybC03LjQ1LTMyLjM2Yy0xLjcyLTcuMTItMy4yMy0xMC4zNS0xNS43My0xNi42QzEyNC44OSw4LjY5LDEwMy43Ni41LDk3LjUxLjUsOTEuNjkuNSw5MCw4LDgzLjA2LDhjLTYuNjgsMC0xMS42NC01LjYtMTcuODktNS42LTYsMC05LjkxLDQuMDktMTIuOTMsMTIuNSwwLDAtOC40MSwyMy43Mi05LjQ5LDI3LjE2QTYuNDMsNi40MywwLDAsMCw0Mi41Myw0NGMwLDkuMjIsMzYuMywzOS40NSw4NC45NCwzOS40NU0xNjAsNzIuMDdjMS43Myw4LjE5LDEuNzMsOS4wNSwxLjczLDEwLjEzLDAsMTQtMTUuNzQsMjEuNzctMzYuNDMsMjEuNzdDNzguNTQsMTA0LDM3LjU4LDc2LjYsMzcuNTgsNTguNDlhMTguNDUsMTguNDUsMCwwLDEsMS41MS03LjMzQzIyLjI3LDUyLC41LDU1LC41LDc0LjIyYzAsMzEuNDgsNzQuNTksNzAuMjgsMTMzLjY1LDcwLjI4LDQ1LjI4LDAsNTYuNy0yMC40OCw1Ni43LTM2LjY1LDAtMTIuNzItMTEtMjcuMTYtMzAuODMtMzUuNzgiLz48L3N2Zz4=
+      mediatype: image/svg+xml
+  keywords:
+    - storage
+  links:
+    - name: Documentation
+      url: https://github.com/openshift/secrets-store-csi-driver-operator
+    - name: Source Repository
+      url: https://github.com/openshift/secrets-store-csi-driver-operator
+  version: 4.14.0
+  maturity: preview
+  maintainers:
+    - email: aos-storage-staff@redhat.com
+      name: Red Hat
+  minKubeVersion: 1.26.0
+  provider:
+    name: Red Hat
+  labels:
+    alm-owner-metering: secrets-store-csi-driver-operator
+    alm-status-descriptors: secrets-store-csi-driver-operator.v4.14.0
+  selector:
+    matchLabels:
+      alm-owner-metering: secrets-store-csi-driver-operator
+  installModes:
+  - type: OwnNamespace
+    supported: false
+  - type: SingleNamespace
+    supported: false
+  - type: MultiNamespace
+    supported: false
+  - type: AllNamespaces
+    supported: true
+  install:
+    strategy: deployment
+    spec:
+      permissions:
```

Comment:
What was the source for this lengthy list of rules below `permissions:` and `clusterPermissions:`?

Reply:
A few difference places:
- The namespace-specific permissions (those under `permissions`) are the same as existing operators like GCP Filestore and is limited to the namespace where the operator is installed (usually `openshift-cluster-csi-drivers`). We know for example the operator must create pods in the same namespace to start the driver.
- The cluster-wide permissions (those under `clusterPermissions`) come from:
  - common permissions required by library-go controllers that are used by most (or all) CSI operators. Some examples of this are ClusterCSIDriver, Leases, VolumeAttachments, and the operator-lock ConfigMap.
  - Roles, RoleBindings, ClusterRoles, and ClusterBindings which are required to grant permissions to the CSI driver.
  - permissions that the operator grants to the driver must first be granted to operator (the operator can not assign permissions that it does not have), so some permissions (like privileged SCC and secretproviderclasses) came from the permissions granted to the driver in [privileged_role.yaml](https://github.com/openshift/secrets-store-csi-driver-operator/blob/main/assets/rbac/privileged_role.yaml) and [secretproviderclasses_role.yaml](https://github.com/openshift/secrets-store-csi-driver-operator/blob/main/assets/rbac/secretproviderclasses_role.yaml).